src/engine/Main.java

```diff
@@ -152,101 +169,182 @@ public static void main(String args[]) {
 		Terminal.print(".");
 
 		game.addWord(new Verb("interact talk speak converse negotiate chat gossip", null, (Object o, Engine t) -> {
-			if (o.alive) {
-				Entity e = (Entity) o;
+			if(o.alive) {
+				Entity e = (Entity)o;
 				e.interaction.accept(t.protag, t);
 			}
 		}));
-
-		Terminal.print(".");
-
-		game.addWord(new Verb("attack assault assail punch hit kick pummel strike kill", null, (Object o, Engine t) -> {
-			o.health -= t.protag.strength;
-			if (o.alive) {
+        
+        Terminal.print(".");
+        
+		game.addWord(new Verb("attack assault assail hit pummel strike kill destroy", null, (Object o, Engine t) -> {
+			if (o.equals(t.protag)) {
+				t.protag.health = 0;
+				Terminal.println("You killed yourself. Nice job.");
+				return;
+			}
+			
+			o.health -= t.protag.strength * t.protag.weapon.damage;
+			t.protag.health -= t.protag.strength * t.protag.weapon.playerDamage;
+			
+			if (!t.protag.weapon.abstractObj)
+				t.protag.weapon.health -= t.protag.strength;
+			
+			if(o.alive) {
 				try {
-					Entity e = (Entity) o;
-					if (e.anger < e.restraint)
-						e.anger = e.restraint;
-				} catch (Exception e) {
-				}
-				;
+				Entity e = (Entity) o;
+				if(e.anger < e.restraint) e.anger = e.restraint;
+				} catch(Exception e) {};
+			}
+			//Terminal.println("You attacked the " + o.accessor + " with the " + t.protag.weapon.accessor + ".");
+			Terminal.println("Weapon: " + t.protag.weapon.accessor);
+		}, (Object o1, Object with, Engine t) -> {
+			t.protag.weapon = with;
+			
+			if (o1.equals(t.protag)) {
+				t.protag.health = 0;
+				Terminal.println("You killed yourself. Nice job.");
+				return;
 			}
-			Terminal.println("You attacked the " + o.accessor + ".");
+			
+			o1.health -= t.protag.strength * t.protag.weapon.damage;
+			t.protag.health -= t.protag.strength * t.protag.weapon.playerDamage;
+			
+			if (!t.protag.weapon.abstractObj)
+				t.protag.weapon.health -= t.protag.strength;
+			
+			if(o1.alive) {
+				try {
+				Entity e = (Entity) o1;
+				if(e.anger < e.restraint) e.anger = e.restraint;
+				} catch(Exception e) {};
+			}
+			//Terminal.println("You attacked the " + o1.accessor + " with the " + t.protag.weapon.accessor + ".");
+			Terminal.println("Weapon: " + with.accessor);
 		}));
 
 		Terminal.print(".");
 
 		game.addWord(new Verb("hold equip", null, (Object o, Engine t) -> {
 			boolean b = o.holdable;
-			if (t.protag.inventory.contains(o)) {
```

Comment:
Why did you delete this?

Reply:
I'm so glad you caught that! I think I was editing the code to test weapons by holding objects, but I deleted it because I didn't like it. I must have not changed everything back accidentally.